pkg/kubernetes/secrets/store.go

```diff
@@ -24,54 +24,56 @@ const (
 
 // Store implements the backing store for secrets as kubernetes secrets.
 type Store struct {
-	logger    hclog.Logger
-	config    config.Config
+	*portercontext.Context
 	hostStore cnabsecrets.Store
+	Secrets   map[string]map[string]string
+	namespace string
 	clientSet *kubernetes.Clientset
+	logger    hclog.Logger
 }
 
-func NewStore(cfg config.Config, l hclog.Logger) cnabsecrets.Store {
+func NewStore(c *portercontext.Context, cfg PluginConfig) *Store {
+	namespace := cfg.Namespace
 	s := &Store{
-		config:    cfg,
-		logger:    l,
-		hostStore: &host.SecretStore{},
+		Secrets:   make(map[string]map[string]string),
+		hostStore: &cnabhost.SecretStore{},
+		namespace: namespace,
+		logger:    cfg.Logger,
 	}
-
-	return portersecrets.NewSecretStore(s)
+	return s
 }
 
-func (s *Store) Connect() error {
+func (s *Store) connect() error {
 
 	if s.clientSet != nil {
 		return nil
 	}
-
-	clientSet, namespace, err := k8s.GetClientSet(s.config.Namespace, s.logger)
-
+	s.logger.Info(fmt.Sprintf("Store.connect: pre-clientset %s : %s", "namespace", s.namespace))
+	clientSet, namespace, err := k8shelper.GetClientSet(s.namespace)
 	if err != nil {
-		s.logger.Debug(fmt.Sprintf("Failed to get Kubernetes Client Set: %v", err))
 		return err
 	}
+	s.namespace = *namespace
+	s.logger.Info(fmt.Sprintf("Store.connect: post-clientset %s : %s", "namespace", s.namespace))
 
 	s.clientSet = clientSet
-	s.config.Namespace = *namespace
 
 	return nil
 }
 
 func (s *Store) Resolve(keyName string, keyValue string) (string, error) {
+	if err := s.connect(); err != nil {
+		return "", err
+	}
 	if strings.ToLower(keyName) != SecretSourceType {
```

Comment:
As we were discussing today, the k8s secret plugin should really only resolve value, secret and perhaps in a later PR, config-map. Can you remove this part which falls back to the host plugin? That will remove support for file, env, command, etc.

Reply:
@carolynvs  We should spell this out in the README then for sure. If the k8s plugin is set then ONLY value, secret (and later config-map) will be supported? Seems like this might be limiting for running this with a local porter instead of the operator is that the behavior we actually want? Maybe I'm not understanding how this would work in that case

Reply:
Does porter fall back to other credSet types if a plugin does not support them?  It's unclear if this behavior change would allow a user to create a credSet with creds of different types.

Reply:
I've only ever thought of using this plugin inside kubernetes. I'm not sure why someone would use it for running porter locally but yeah that would cause trouble if we didn't embed the fallback logic directly into the plugin since Porter doesn't try other plugins if the first one can't resolve it. There's only one secret plugin active at a time.

It doesn't hurt anything to leave the host plugin in there.